From 8026fa014d4f6243c2893e9df580b3c47ea656a9 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Sun, 11 Sep 2022 11:30:28 +0300
Subject: [PATCH] [SW-0] erom: remove pcie_init dead code

PCIe init in erom code is done as part of secondary main
in arc0, in function pcie_config().
PCIe enabling in EROM is not the same as PCIe enable in
preboot, as it only enables GEN1 and not GEN3 PCIe.
This patch removes other related pcie init functions,
which are not required nor being called, and may only
confuse when reading the code.

Change-Id: I41b5338921a5dea232a3ea2b942617fedf386a99
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/app/ext_rom/erom_pcie.c b/app/ext_rom/erom_pcie.c
index f60526b..a47c627 100644
--- a/app/ext_rom/erom_pcie.c
+++ b/app/ext_rom/erom_pcie.c
@@ -71,69 +71,6 @@
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_PCIE_PHY_RST_N);
 }
 
-static void preboot_pcie_device_id_vendor_id_set(void)
-{
-	u32 device_id_vendor_id = hl_pci_get_ven_id_dev_id();
-
-	/* Update Vendor ID */
-	regmap_writel(device_id_vendor_id,
-		      mmPCIE_DBI_DEVICE_ID_VENDOR_ID_REG);
-
-	/* Update subsystem Vendor ID */
-	regmap_writel(device_id_vendor_id,
-		      mmPCIE_DBI_SUBSYSTEM_ID_SUBSYSTEM_VENDOR_ID_REG);
-
-}
-
-void erom_init_pcie(void)
-{
-	u32 link_control;
-
-	preboot_init_pcie_phy();
-
-	preboot_pcie_device_id_vendor_id_set();
-
-	/* Execute the Synopsys W/A */
-	regmap_writel(0xF, mmPCIE_PHY_CR_PARA_CTRL);
-	regmap_writel(0x0004A07D, mmPCIE_PHY_CR_PARA_CMD);
-	udelay(20);
-
-	/* Set HBM BAR size */
-	preboot_update_bar_sizes();
-
-	/* PCIe disable LTR */
-	preboot_disable_pcie_ltr();
-
-	link_control = regmap_readl(mmPCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG);
-	link_control &= 0xFFFFFFF0;
-	link_control |= 1;
-	regmap_writel(link_control, mmPCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG);
-
-	/* Enable and Mask LTSSM (erratum 102) */
-	regmap_writel(3, mmPCIE_AUX_LTSSM_EN);
-}
-
-void erom_pcie_enable(void)
-{
-	u32 rst_src;
-
-	rst_src = hl_rst_src_get();
-	switch (rst_src) {
-	case HL_COLD_RST:
-	case HL_PRSTN_RST:
-		/* PCIe is reset, so must initialize PCIe */
-		erom_init_pcie();
-		break;
-	default:
-		/*
-		 * Currently, keep using the same flow as done before,
-		 * i.e. PCIe should already be initialized by default.
-		 * Except COLD & PRSTN resets, keep PCIe intact.
-		 */
-		break;
-	}
-}
-
 static bool hl_pcie_link_up(void)
 {
 	unsigned int prstn, smlh, rdlh, rst_src;
